windows-agent/internal/proservices/certs.go

```diff
@@ -0,0 +1,61 @@
+package proservices
+
+import (
+	"crypto/rand"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"math/big"
+
+	"github.com/canonical/ubuntu-pro-for-wsl/common"
+	"github.com/canonical/ubuntu-pro-for-wsl/common/certs"
+	"github.com/ubuntu/decorate"
+)
+
+// newTLSCertificates creates a self-signed root CA, agent and clients certificates and writes them into destDir.
+func newTLSCertificates(destDir string) (c agentCerts, err error) {
+	decorate.OnError(&err, "could not create TLS credentials:")
+
+	// generates a pseudo-random serial number for the root CA certificate.
+	serial, err := rand.Int(rand.Reader, new(big.Int).Lsh(big.NewInt(1), 128))
+	if err != nil {
+		return agentCerts{}, fmt.Errorf("failed to generate serial number for the CA cert: %v", err)
+	}
+
+	rootCert, rootKey, err := certs.CreateRootCA(common.GRPCServerNameOverride, serial, destDir)
+	if err != nil {
+		return agentCerts{}, err
+	}
+
+	// Create and write the server and client certificates signed by the root certificate created above.
+	serverCert, err := certs.CreateTLSCertificateSignedBy("server", common.GRPCServerNameOverride, serial.Rsh(serial, 2), rootCert, rootKey, destDir)
```

Comment:
So, here too, I would clarify in the filename the whole terminology:
- rootCA
- agent certificate
- client certificate

(ensure to align all the tests around those name).

That way, looking at the FHS, it’s clear what we are loading where.